extensions/integration-tests/testfmks/src/stress.ts

```diff
@@ -0,0 +1,289 @@
+		//
+		await Promise.all(pendingPromises);
+
+		let total = numPasses + errors.length + fails.length;
+		assert(numPasses >= passThreshold * total, `Call Stressified: ${functionName}(${args.join(',')}) failed with a expected pass percent of ${passThreshold * 100}, actual pass percent is: ${numPasses * 100 / total}`);
+		return { numPasses: numPasses, fails: fails, errors: errors };
+	}
+}
+
+// the singleton Stress object.
+//
+const stresser = new Stress();
+
+/**
+ * Decorator Factory to return the Method Descriptor function that will stressify any test class method.
+		* Using the descriptor factory allows us pass options to the discriptor itself separately from the arguments
+		* of the function being modified.
+ * @param runtime - The desconstructed {@link StressOptions} option. see {@link StressOptions} for details.
+ * @param dop - The desconstructed {@link StressOptions} option. see {@link StressOptions} for details.
+ * @param iterations - The desconstructed {@link StressOptions} option. see {@link StressOptions} for details.
+ * @param passThreshold - The desconstructed {@link StressOptions} option. see {@link StressOptions} for details.
+ */
+export function stressify({ runtime, dop, iterations, passThreshold }: StressOptions = DefaultStressOptions): (memberClass: any, memberName: string, memberDescriptor: PropertyDescriptor) => PropertyDescriptor {
+	// return the function that does the job of stressifying a test class method with decorator @stressify
+	//
+	debug(`stressify FactoryDecorator called with runtime=${runtime}, dop=${dop}, iter=${iterations}, passThreshold=${passThreshold}`);
+
+	// The actual decorator function that modifies the original target method pointed to by the memberDiscriptor
+	return function (memberClass: any, memberName: string, memberDescriptor: PropertyDescriptor): PropertyDescriptor {
+		// stressify the target function pointed to by the descriptor.value only if
+		// SuiteType is stress
+		//
+		const suiteType = getSuiteType();
+		debug(`Stressified Decorator called for: ${memberName} and suiteType=${suiteType}`);
+		if (suiteType === SuiteType.Stress) {
+			// save a reference to the original method
+			// this way we keep the values currently in the
+			// descriptor and don't overwrite what another
+			// decorator might have done to the descriptor.
+			debug(`Stressifying ${memberName} since env variable SuiteType is set to ${SuiteType.Stress}`);
+			const originalMethod: Function = memberDescriptor.value;
+			//editing the descriptor/value parameter
+			memberDescriptor.value = async function (...args: any[]): Promise<StressResult> {
+				// note usage of originalMethod here
+				//
+				assert(stresser !== null && stresser !== undefined, 'stresser object must be defined');
+				const result: StressResult = await stresser.run(originalMethod, this, memberName, args, { runtime, dop, iterations, passThreshold });
```

Comment:
I guess it doesn't hurt but if this is ever not defined something has seriously gone wrong. I don't think it's necessary

Reply:
Yes removing.